### PR TITLE
(PXP-7760) Feat/remove empty field lists

### PR DESCRIPTION
**Pull request by ocshawn**
Jira Ticket: [PXP-7760](https://ctds-planx.atlassian.net/browse/PXP-7760)

### New Features
- add hideEmptyFilterSection flag and functionality to hide empty filter sections 
- add handling of edge case where tab will have no visible filters
- update FilterSection tests to test new hideEmptyFilterSection flag

### CODE REVIEW

File: src/components/filters/FilterList/index.jsx [hunk 1]
```diff
@@ -85,47 +85,64 @@ class FilterList extends React.Component {
...
+            (...args) => this.handleDragRangeFilter(index, ...args)
+          }
+          hideZero={this.props.hideZero}
+          tierAccessLimit={this.props.tierAccessLimit}
+          lockedTooltipMessage={this.props.lockedTooltipMessage}
+          disabledTooltipMessage={this.props.disabledTooltipMessage}
+        />,
+      );
+    });
+    // if no filters in tab add meassage
```
Comment (williamhaley) [inline src/components/filters/FilterList/index.jsx]:
```suggestion
    // if no filters in tab add message
```

File: src/components/filters/FilterList/index.jsx [hunk 2]
```diff
@@ -85,47 +85,64 @@ class FilterList extends React.Component {
     // Takes in parent component's filterStatus or self state's filterStatus
     const filterStatus = this.props.filterStatus
       ? this.props.filterStatus : this.state.filterStatus;
-
+    const filterSectionToShow = [];
+    this.props.sections.forEach((section, index) => {
```
Comment (williamhaley) [inline src/components/filters/FilterList/index.jsx:89]: We could change this from a `forEach` to a `map` like...

```
const filterSectionToShow = this.props.sections.map((section, index) => {
...
return (
  <FilterSection
...
```

Unless some side-effect logic is happening in the loop that I'm not seeing 🔍

File: src/components/filters/FilterList/index.jsx [hunk 3]
```diff
@@ -85,47 +85,64 @@ class FilterList extends React.Component {
...
+        />,
+      );
+    });
+    // if no filters in tab add meassage
+    if (filterSectionToShow.length === 0) {
+      filterSectionToShow.push(
+        <div key='g3-filter-list-empty' className='g3-filter-list-empty g3-filter-section'>
+          <h4>
+            Selected Data Does NOT Include
+            <br />
```
Comment (williamhaley) [inline src/components/filters/FilterList/index.jsx:136]: I think for html5 `<br>` [is fine](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/br)

**Review by williamhaley — APPROVED** [review]
lgtm! :shipit:

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@gen3/ui-component",
-  "version": "0.6.6",
+  "version": "0.6.7",
```
Comment (mfshao) [inline package.json]: bump up minor version if you think this is a new feature, otherwise, keep this patch version but change what you have in PR description to `Improvements`

**Comment by mfshao** [comment]
I'd suggest to let @ZakirG take a look so he is aware what is it going to change since he might be also working on the same component but for a different feature

File: src/components/filters/FilterList/index.jsx [hunk 5]
```diff
@@ -85,47 +85,64 @@ class FilterList extends React.Component {
...
+    // if no filters in tab add message
+    if (filterSectionToShow.length === 0) {
+      filterSectionToShow.push(
+        <div key='g3-filter-list-empty' className='g3-filter-list-empty g3-filter-section'>
+          <h4>
+            Selected Data Does NOT Include
+            <br />
+            These Filters
+          </h4>
+          <span>Disable Filters in other tabs to view</span>
```
Comment (mfshao) [inline src/components/filters/FilterList/index.jsx]: What if, say, data explorer is in `private` mode, and a user with no access to anything logged in? Would he/she just see a bunch of empty filter lists with this particular message? If so, lets remove this sentence, because it will be confusing to users

**Review by mfshao — COMMENTED** [review]
I would prefer to update this feature into one of the stories in the `gen3-ui-component`'s storybook 😛

File: src/components/filters/FilterList/index.jsx (hunk 2 above)
Comment (ocshawn) [reply src/components/filters/FilterList/index.jsx:89]: i will revisit, map will not work due to below if statement, but reduce will probably work

Comment (ocshawn) [reply src/components/filters/FilterList/index.jsx:89]: do you have any thoughts? i think forEach may be the best way to do this

File: src/components/filters/FilterList/index.jsx (hunk 3 above)
Comment (ocshawn) [inline src/components/filters/FilterList/index.jsx:136]:
```suggestion
            <br>
```

File: src/components/filters/FilterList/index.jsx (hunk 2 above)
Comment (williamhaley) [reply src/components/filters/FilterList/index.jsx:89]: Ah, yeah, I see. `forEach` works! Another way to do it would be to return `null` and filter (edit) _or use reduce like you suggested_, but at that point the `forEach` is probably more readable. Looks good as is!

File: src/components/filters/FilterList/index.jsx (hunk 5 above)
Comment (ocshawn) [reply src/components/filters/FilterList/index.jsx]:
```suggestion
```

File: src/components/filters/FilterList/index.jsx (hunk 3 above)
Comment (ocshawn) [reply src/components/filters/FilterList/index.jsx:136]: FYI jsx still needs the backslash, otherwise our tests fail

File: package.json (hunk 4 above)
Comment (ocshawn) [reply package.json]: updated to 0.7.0 also added to the storybook

File: src/components/filters/FilterList/index.jsx (hunk 3 above)
Comment (williamhaley) [reply src/components/filters/FilterList/index.jsx:136]: 👍 Sorry for the wasted cycles on that

**Review by mfshao — APPROVED** [review]
👍
